File: integration-tests-java/src/test/java/io/micronaut/jimmer/java/it/config/TestConsumerZone.java
package io.micronaut.jimmer.java.it.config;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import java.time.ZoneId;
import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.runtime.JSqlClientImplementor;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
public class TestConsumerZone {

    @Inject JSqlClient jSqlClient;

    @Inject
    @Named(Constant.DATASOURCE2)
    JSqlClient jSqlClientDB2;

    @Test
    void testConsumer() {
        JSqlClientImplementor defaultJSqlClientImplementor = (JSqlClientImplementor) jSqlClient;
        ZoneId zoneId = defaultJSqlClientImplementor.getZoneId();

        JSqlClientImplementor jSqlClientImplementorDB2 = (JSqlClientImplementor) jSqlClientDB2;
        ZoneId zoneIdDB2 = jSqlClientImplementorDB2.getZoneId();

        Assertions.assertEquals(zoneId, ZoneId.of("Asia/Shanghai"));
        Assertions.assertEquals(zoneIdDB2, ZoneId.of("Asia/Chongqing"));
    }
}

File: integration-tests-kotlin/src/test/kotlin/io/micronaut/jimmer/kotlin/it/config/TestConsumerZone.kt
package io.micronaut.jimmer.kotlin.it.config

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import jakarta.inject.Named
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.time.ZoneId

@MicronautTest
class TestConsumerZone {
    @Inject
    lateinit var kSqlClient: KSqlClient

    @Inject
    @Named(Constant.DATASOURCE2)
    lateinit var kSqlClientDB2: KSqlClient

    @Test
    fun testConsumer() {
        val defaultJSqlClientImplementor = kSqlClient.javaClient
        val zoneId = defaultJSqlClientImplementor.zoneId

        val jSqlClientImplementorDB2 = kSqlClientDB2.javaClient
        val zoneIdDB2 = jSqlClientImplementorDB2.zoneId

        Assertions.assertEquals(zoneId, ZoneId.of("Asia/Shanghai"))
        Assertions.assertEquals(zoneIdDB2, ZoneId.of("Asia/Chongqing"))
    }
}
